### PR TITLE
fix(iast): partial matches on function names should not be patched [backport 2.11]

### PR DESCRIPTION
**Pull request by gnufede**
backport of #11015
Code Security: This PR fixes an issue where partial matches of the functions we want to patch would be patched instead of exact matches
## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance
policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

(cherry picked from commit a72b931245ba24a30687d8c31c257777332238ae)

### CODE REVIEW

File: tests/appsec/iast/_ast/test_ast_patching.py [hunk 1]
```diff
@@ -155,3 +155,18 @@ def test_module_path_none(caplog):
...
+@pytest.mark.parametrize(
+    "module_name",
+    [
+        ("tests.appsec.iast.fixtures.ast.other.globals_builtin"),
+    ],
+)
+def test_astpatch_globals_module_unchanged(module_name):
+    module_path, new_source = astpatch_module(__import__(module_name, fromlist=[None]))
+    assert ("", "") == (module_path, new_source)
+    if ("", "") != (module_path, new_source):
```
Comment (christophe-papazian) [inline tests/appsec/iast/_ast/test_ast_patching.py:169]: didn't we assert the line before that it could not happen?

Comment (gnufede) [reply tests/appsec/iast/_ast/test_ast_patching.py:169]: Yes indeed

Comment (gnufede) [reply tests/appsec/iast/_ast/test_ast_patching.py:169]: But it says very little about the expected result.

The lines below are the ones we actually don't want to happen

Comment (christophe-papazian) [reply tests/appsec/iast/_ast/test_ast_patching.py:169]: But if the assert failed, it breaks out of the test function, so the if condition is never true.

**Review by christophe-papazian — APPROVED** [review]

**Review by avara1986 — APPROVED** [review]

**Pull request merged**